### PR TITLE
Feature/guided tour

### PR DESCRIPTION
**Pull request by pvilalta**
https://platform-dev-guided-tour.bimdata.io

Hello team,

Here is the initial version of the guided tour. It's integrated in the platform without any condition, thats mean each time you will start the platform or refresh it, the guided tour will occur.

What I have done

* created a specific component for the first and last step
* made up a basic scenario to use the component

things to improve

* ~manage if 2 or more tours are available at the same time~
* ~manage if the user has already watched a tour~
* ~flag a specific element with 'data-guide" that come from an iteration (I made a simple implementation here, that spotlight the first element only)~

This PR is related with the one on the design-system that I also pushed
https://github.com/bimdata/design-system/pull/173

To test the feature you have to pull the branch from the design-system and either

* build and copy/paste dist file into the platform
or
* build and create a relation between the platform and the design-system ("@bimdata/design-system": "file:../design-system")

It's not the intent to merge this PR in develop currently, more update will occur over time (through sub PR)

### CODE REVIEW

**Review by NicolasRichel — APPROVED** [review]

File: src/state/user.js [hunk 1]
```diff
@@ -24,6 +26,21 @@ const loadUser = async () => {
...
+  return guidedTour.json();
+};
+
+const completedTour = async tour => {
+  const { accessToken } = useAuth();
+  const guidedTour = await PlatformService.completedTour(
+    accessToken.value,
+    tour
+  );
+  return guidedTour.json();
```
Comment (NicolasRichel) [inline src/state/user.js]: I think the `.json()` deserialization should be done in the service rather than the state.

**Review by NicolasRichel — CHANGES_REQUESTED** [review]

File: src/services/PlatformService.js [hunk 2]
```diff
@@ -11,6 +11,26 @@ class PlatformService {
...
+    return fetch(`${process.env.VUE_APP_BACKEND_BASE_URL}/guidedtour/`, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${accessToken}`
+      }
+    });
+  }
+
+  completedTour(accessToken, tour) {
```
Comment (Amoki) [inline src/services/PlatformService.js]: What about `setTourCompleted` ?

**Review by Amoki — CHANGES_REQUESTED** [review]

**Review by NicolasRichel — APPROVED** [review]

File: src/services/PlatformService.js [hunk 3]
```diff
@@ -11,6 +11,33 @@ class PlatformService {
...
+          Authorization: `Bearer ${accessToken}`
+        }
+      }
+    );
+
+    return res.json();
+  }
+
+  async setTourCompleted(accessToken, tour) {
+    const res = fetch(`${process.env.VUE_APP_BACKEND_BASE_URL}/guidedtour/`, {
```
Comment (Amoki) [inline src/services/PlatformService.js]:
```suggestion
    const res = await fetch(`${process.env.VUE_APP_BACKEND_BASE_URL}/guidedtour/`, {
```

File: src/state/user.js [hunk 4]
```diff
@@ -24,6 +26,16 @@ const loadUser = async () => {
   return user;
 };
 
+const loadGuidedTours = async () => {
+  const { accessToken } = useAuth();
```
Comment (Amoki) [inline src/state/user.js]: AccessToken can be managed directly in the service, see https://github.com/bimdata/platform-next/blob/develop/src/services/BcfService.js#L32

**Review by Amoki — CHANGES_REQUESTED** [review]

**Review by Amoki — APPROVED** [review]

**Review by NicolasRichel — APPROVED** [review]

**Review by NicolasRichel — APPROVED** [review]

**Pull request merged**